packages/story-editor/src/app/api/apiProvider.js

```diff
@@ -442,88 +125,116 @@ function APIProvider({ children }) {
     [cdnURL, assetsURL]
   );
 
-  const getCustomPageTemplates = useCallback(
-    (page = 1) => {
-      let apiPath = customPageTemplates;
-      const perPage = 100;
-      apiPath = addQueryArgs(apiPath, {
-        context: 'edit',
-        per_page: perPage,
-        page,
-        _web_stories_envelope: true,
-      });
-      return apiFetch({ path: apiPath }).then(({ headers, body }) => {
-        const totalPages = parseInt(headers['X-WP-TotalPages']);
-        const templates = body.map((template) => {
-          return { ...template['story_data'], templateId: template.id };
-        });
-        return {
-          templates,
-          hasMore: totalPages > page,
-        };
-      });
-    },
-    [customPageTemplates]
-  );
-
-  const addPageTemplate = useCallback(
-    (page) => {
-      return apiFetch({
-        path: `${customPageTemplates}/`,
-        data: {
-          story_data: page,
-          status: 'publish',
-        },
-        method: 'POST',
-      }).then((response) => {
-        return { ...response['story_data'], templateId: response.id };
-      });
-    },
-    [customPageTemplates]
-  );
-
-  const deletePageTemplate = useCallback(
-    (id) => {
-      // `?_method=DELETE` is an alternative solution to override the request method.
-      // See https://developer.wordpress.org/rest-api/using-the-rest-api/global-parameters/#_method-or-x-http-method-override-header
-      return apiFetch({
-        path: addQueryArgs(`${customPageTemplates}${id}/`, {
-          _method: 'DELETE',
-        }),
-        data: { force: true },
-        method: 'POST',
-      });
-    },
-    [customPageTemplates]
-  );
-
-  const state = {
-    actions: {
-      autoSaveById,
-      getStoryById,
-      getDemoStoryById,
-      getStoryLockById,
-      setStoryLockById,
-      deleteStoryLockById,
-      getMedia,
-      getLinkMetadata,
-      saveStoryById,
-      getAuthors,
-      uploadMedia,
-      updateMedia,
-      deleteMedia,
-      saveMetaBoxes,
-      getStatusCheck,
-      addPageTemplate,
-      getCustomPageTemplates,
-      deletePageTemplate,
-      getPageTemplates,
-      getCurrentUser,
-      updateCurrentUser,
-    },
-  };
-
-  return <Context.Provider value={state}>{children}</Context.Provider>;
+  // @todo Handle undefined actions in their components.
```

Comment:
Can you elaborate on that?

Reply:
When some callbacks are not provided, let's say by the playground integration, it should not throw errors and I would add checks to handle these `undefined` callbacks in the components they have been used in.